sdk/communication/AzureCommunication/Source/CommunicationIdentifierSerializer.swift

```diff
@@ -99,6 +72,55 @@ public class CommunicationIdentifierSerializer {
 
         return CommunicationCloudEnvironment(environmentValue: model.requestString);
     }
+    
+    static func assertOneNestedModel(_ identifier: CommunicationIdentifierModel) throws {
+        var presentProperties = [String]()
+        
+        if let _ = identifier.communicationUser {
+            presentProperties.append("communicationUser")
+        }
+        if let _ = identifier.phoneNumber {
+            presentProperties.append("phoneNumber")
+        }
+        if let _ = identifier.microsoftTeamsUser {
+            presentProperties.append("microsoftTeamsUser")
+        }
+        
+        if presentProperties.count > 1 {
+            throw AzureError.client("Only one property should be present")
+        }
+    }
+
+    static func serialize(identifier: CommunicationIdentifier) throws -> CommunicationIdentifierModel {
+        switch identifier {
+        case let user as CommunicationUserIdentifier:
+            return CommunicationIdentifierModel(rawId: nil,
+                                                communicationUser: CommunicationUserIdentifierModel(id: user.identifier),
+                                                phoneNumber: nil,
+                                                microsoftTeamsUser: nil)
+        case let phoneNumber as PhoneNumberIdentifier:
+            return CommunicationIdentifierModel(rawId: phoneNumber.rawId,
+                                                communicationUser: nil,
+                                                phoneNumber: PhoneNumberIdentifierModel(value: phoneNumber.phoneNumber),
+                                                microsoftTeamsUser: nil)
+        case let teamsUser as MicrosoftTeamsUserIdentifier:
+            return try CommunicationIdentifierModel(rawId: teamsUser.rawId,
+                                                    communicationUser: nil,
+                                                    phoneNumber: nil,
+                                                    microsoftTeamsUser:
+                                                        MicrosoftTeamsUserIdentifierModel(userId: teamsUser.userId,
+                                                                                          isAnonymous: teamsUser.isAnonymous,
+                                                                                          cloud: serialize(cloud: teamsUser.cloudEnviroment)))
+        case let unknown as UnknownIdentifier:
+            return CommunicationIdentifierModel(rawId: unknown.identifier,
+                                                communicationUser: nil,
+                                                phoneNumber: nil,
+                                                microsoftTeamsUser: nil)
+        default:
+            throw AzureError.client("Not support kind in CommunicationIdentifier.")
```

Comment:
should we rename the 'kind' in the error string? Since 'kind' concept is removed in this PR

Reply:
Oh good catch